### ISSUE TITLE
ReadsPipelineSpark: null point exception during MannWhitney permutation test

### ISSUE THREAD

**Issue by frank-y-liu** [broadinstitute/gatk#4265]
When running a large genome data set (`WGS-G94982-NA12878-no-NC_007605.bam`), the haplotypeCallerSpark raised an Null pointer exception in MannWhitney permutation test. Trace:

```
18/01/22 17:11:46 WARN scheduler.TaskSetManager: Lost task 161.0 in stage 26.0 (TID 110570, 192.168.100.210, executor 75): java.lang.NullPointerException
        at org.broadinstitute.hellbender.utils.MannWhitneyU.permutationTest(MannWhitneyU.java:537)
        at org.broadinstitute.hellbender.utils.MannWhitneyU.test(MannWhitneyU.java:409)
        at org.broadinstitute.hellbender.tools.walkers.annotator.RankSumTest.annotate(RankSumTest.java:75)
        at org.broadinstitute.hellbender.tools.walkers.annotator.VariantAnnotatorEngine.annotateContext(VariantAnnotatorEngine.java:266)
        at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCallerGenotypingEngine.makeAnnotatedCall(HaplotypeCallerGenotypingEngine.java:298)
        at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCallerGenotypingEngine.assignGenotypeLikelihoods(HaplotypeCallerGenotypingEngine.java:148)
        at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCallerEngine.callRegion(HaplotypeCallerEngine.java:566)
        at org.broadinstitute.hellbender.tools.HaplotypeCallerSpark.lambda$regionToVariants$2(HaplotypeCallerSpark.java:253)
        at org.broadinstitute.hellbender.tools.HaplotypeCallerSpark$$Lambda$312.00000000EC650830.apply(Unknown Source)
        at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:278)
        at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1823)
        at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:305)
        at java.util.stream.StreamSpliterators$WrappingSpliterator$$Lambda$314.00000000EC651070.getAsBoolean(Unknown Source)
        at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:217)
        at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:180)
        at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:311)
        at java.util.Spliterators$1Adapter.hasNext(Spliterators.java:692)
        at scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:42)
        at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:215)
        at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:957)
        at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:948)
        at org.apache.spark.storage.BlockManager.doPut(BlockManager.scala:888)
        at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:948)
        at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:694)
        at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:334)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:285)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
        at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
        at org.apache.spark.scheduler.Task.run(Task.scala:99)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:282)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(Thread.java:785)

```
No vcf file was created. 

So far it seems walker mode doesn't have this issue.

**Comment by frank-y-liu** [comment]
The issue is related to the input data. When `CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam` was used as the input file, the run was successful.

**Comment by lbergelson** [comment]
@frank-y-liu Thanks for reporting these.

**Comment by frank-y-liu** [comment]
One more data point: running `NA12878.ga2.exome.maq.raw.bam` caused the exception. Input files used:
`-I NA12878.ga2.exome.maq.raw.bam`, `-R Homo_sapiens_assembly18.2bit`, `--knownSites dbsnp_138.hg18.vcf`

**Comment by lbergelson** [comment]
Thanks, that's helpful.

**Comment by lbergelson** [comment]
added another ticket to track this #4290

**Comment by davidbenjamin** [comment]
It looks like this was closed by #5371.
